### PR TITLE
Adding loading indication on Login page

### PR DESCRIPTION
**Pull request by yvmarkovska**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1112

**Summary of the changes made**:
1. Added a state variable to useAuth hook 
2. Importing SpinningLoader to LoginForn and conditional rendering
3. Moving reloading and navigation from Login page in a separate useEffect to prevent flickering of button + additional logic for setting isLogginging state variable
4. Handling null fields error coming from the BE
5. Adding FE validations to check if fields' values have the amount of characters and only allowed type of characters

### CODE REVIEW

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-auth.tsx [hunk 1]
```diff
@@ -198,25 +202,51 @@ const AuthProvider = ({ children }: IAuthProviderProps) => {
...
     }, [ authInfo, getUserFromResponse, isGetAuthInfoSuccess, setUserDetails ]);
 
+    // Purpose: Added to prevent flickering Login button while loading
+    // If the user is successfully logged in > we first navigate to home
+    // then refresh the window to set internal user and header info
+    useEffect(() => {
+        if (!isLoggingIn &&
+            loginSubmitStatus === HttpStatus.Success &&
+            location.pathname === getLoginPath()) {
+            navigate('/');
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-auth.tsx:212]: Clicking the login button on an empty form redirects to home

**Review by shsimeonova — CHANGES_REQUESTED** [review]

**Review by shsimeonova — APPROVED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-auth.tsx (hunk 1 above)
Comment (yvmarkovska) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-auth.tsx:221]: Fixed with validations on the FE and BE

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.tsx [hunk 2]
```diff
@@ -1,44 +1,108 @@
...
+            setDisableLoginButton(true);
+            return;
+        }
+
+        setLoginErrorMessage('');
+        setDisableLoginButton(false);
+    }, [ usernameFormError, passwordFormError, setLoginErrorMessage, hasPressedLoginBtn ]);
+
     const handleLoginClick = useCallback(async () => {
         /* TODO:  Add message to notify the admin if SULS is not working.
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.tsx:95]: Is this implemented?

**Review by shsimeonova — APPROVED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.tsx (hunk 2 above)
Comment (yvmarkovska) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/auth/LoginForm.tsx:95]: Not implemented. Will be implemented when Redux is introduced to this page.

**Review by shsimeonova — APPROVED** [review]

**Pull request merged**